src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -231,16 +231,24 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
     });
   };
 
-  const helperGetChannelDetail = async (channel: ChannelList) => {
+  const helperGetChannelDetail = async (channel: ChannelList, lastChatDate?: string) => {
     if (!localDb) return;
 
     try {
       let response;
       if (channel?.channelType === 'ANON_PM') {
-        response = await AnonymousMessageRepo.getAnonymousChannelDetail('messaging', channel?.id);
+        response = await AnonymousMessageRepo.getAnonymousChannelDetail(
+          'messaging',
+          channel?.id,
+          lastChatDate
+        );
       } else {
         const channelType = channel?.channelType === 'GROUP' ? 'group' : 'messaging';
-        response = await SignedMessageRepo.getSignedChannelDetail(channelType, channel?.id);
+        response = await SignedMessageRepo.getSignedChannelDetail(
+          channelType,
+          channel?.id,
+          lastChatDate
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are adding an optional `lastChatDate` parameter to the `helperGetChannelDetail` function and passing it to the `getAnonymousChannelDetail` and `getSignedChannelDetail` functions. This is a good change as it allows for more flexibility when fetching channel details, potentially allowing for more efficient data fetching if only messages after a certain date need to be fetched. However, there's no error handling in case `lastChatDate` is not a valid date string. It would be better to add some validation or error handling for this.
```diff
+    const helperGetChannelDetail = async (channel: ChannelList, lastChatDate?: string) => {
+       if (!localDb) return;
+   
+       try {
+         let response;
+         if (channel?.channelType === 'ANON_PM') {
+          response = await AnonymousMessageRepo.getAnonymousChannelDetail(
+            'messaging',
+            channel?.id,
+            lastChatDate
+          );
+         } else {
+           const channelType = channel?.channelType === 'GROUP' ? 'group' : 'messaging';
+          response = await SignedMessageRepo.getSignedChannelDetail(
+            channelType,
+            channel?.id,
+            lastChatDate
+          );
+        }
+        helperSaveChannelDetail(channel, response);
+      } catch (e) {
+        // Add error handling here
+      }
+    };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->